src/App.js

```diff
@@ -1,19 +1,31 @@
-import React, {useState} from 'react';
+import React, {useState, useEffect} from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
+function useSemiPersistentState() {
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('savedTodoList')) || []
+  );
+
+  useEffect(() => {
+    localStorage.setItem('savedTodoList', JSON.stringify(todoList));
+  }, [todoList])
+
+  return [todoList, setTodoList];
+}
+
```

Comment:
Yeah the assignment only sortof halfway conveys why it would be useful to make a custom hook like this. I'll go over this a bit more today and we can see if it'll make more sense then